### DISCUSSION TITLE
Do we have a general method for getting the number of basis coef?

### DISCUSSION THREAD

**Discussion by garrettwrong** [ComputationalCryoEM/ASPIRE-Python#674] [Q&A] [answered]
Do we have a consistent method for querying our `Basis` classes for their expected number of coefficients?

I have one or two places in the code where I would like to check user supplied arguments are not larger than the number of basis coefficients (or some more complicated relationship).  I'm realizing now this (`f(resolution): return n_coefs`) would be a useful class method.  Took a quick look, but I didn't see it.  Figured I'd ask in Q&A since other people have become familiar with our basis code and have been working on improvements.

Thanks!

**Comment by garrettwrong** [comment]
To clarify, I see `count` but this requires actually instantiating a class first... :)

**Comment by janden** [comment, marked as answer]
There's no current way of doing this without instantiating a class. Having a class method perform this might be tricky without duplicating a lot of code. I guess the question is, why not instantiate a class an using `count`? Speed?

**Reply by garrettwrong** [reply]
There are a few places we might want to know how many basis functions we would have, but potentially not have a need for a basis at that point in the code.  It's more a curiosity about whether we had it, or if it was involved... Thanks!
